analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PropertiesAccessCorrectField.cs

```diff
@@ -768,34 +786,43 @@ public class ContainsConstraint
 
         public ContainsConstraint IgnoreCase
         {
-            // Refactor this getter so that it actually refers to the field '_ignoreCase'.
-            // _ignoreCase has a different type than the property return type
-            get // Noncompliant - FP
+            get // Compliant - _IgnoreCase has different type
             {
                 _ignoreCase = true;
                 return this;
             }
         }
     }
 
-    public class Base
-    {
-        protected long writeLockTimeout;
-    }
-
-    public class SubClass : Base
+    public class AClass
     {
         public static long WRITE_LOCK_TIMEOUT = 1000;
+        public long longValue;
         public long WriteLockTimeout
         {
             get
             {
-                return writeLockTimeout; // Noncompliant - FP
+                return longValue; // Compliant - WRITE_LOCK_TIMEOUT is static field when the property is not static.
             }
             set
             {
-                this.writeLockTimeout = value; // Noncompliant - FP
+                longValue = value; // Compliant - WRITE_LOCK_TIMEOUT is static field when the property is not static.
+            }
+        }
+
+        public static long TEST_STATIC_CASE = 1000;
+        public static long ALong = 2000;
+        public static long TestStaticCase
+        {
+            get
+            {
+                return ALong; // Noncompliant
+            }
+            set
+            {
+                ALong = value; // Noncompliant
             }
```

Comment:
I think you can improve the coverage with the case when the property is static but the field is not.